### PR TITLE
docs(behavior_path_planner): add pull over out separated docs and update them for freespace

### PR DESCRIPTION
**Pull request by kosuke55**
## Description

Add pull_over/out separeted docs and update for freespace parking.

## Related links

https://github.com/autowarefoundation/autoware.universe/pull/2879

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: planning/behavior_path_planner/behavior_path_planner_pull_out-design.md [hunk 1]
```diff
@@ -0,0 +1,131 @@
+# PullOut Module
```
Comment (satoshi-ota) [inline planning/behavior_path_planner/behavior_path_planner_pull_out-design.md:1]:
```suggestion
# Pull Out design
```

Please align title `Hoge design` :pray: 
https://github.com/autowarefoundation/autoware.universe/pull/2956

File: planning/behavior_path_planner/behavior_path_planner_pull_out-design.md [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+
+Pull out distance is calculated by the speed, lateral deviation, and the lateral jerk. The lateral jerk is searched for among the predetermined minimum and maximum values, and the one that generates a safe path is selected.
+
+- Generate the shoulder lane centerline and shift it to the current position.
+- In the section between merge start and end, path is shifted by a method that is used to generate avoidance path (four segmental constant jerk polynomials)
+- Combine this path with center line of road lane
+
+![shift_pull_out](./image/shift_pull_out.drawio.svg)
+
+[shift pull out vidoe](https://user-images.githubusercontent.com/39142679/187872468-6d5057ee-e039-499b-afc7-fe0dc8052a6b.mp4)
```
Comment (satoshi-ota) [inline planning/behavior_path_planner/behavior_path_planner_pull_out-design.md:82]: typo

File: planning/behavior_path_planner/behavior_path_planner_pull_out-design.md [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+![geometric_pull_out](./image/geometric_pull_out.drawio.svg)
+
+[geometric pull out video](https://user-images.githubusercontent.com/39142679/181024707-3e7ca5ee-62de-4334-b9e9-ded313de1ea1.mp4)
+
+###### parameters for geometric pull out
+
+| Name                        | Unit  | Type   | Description                                             | Default value |
+| :-------------------------- | :---- | :----- | :------------------------------------------------------ | :------------ |
+| enable_geometric_pull_out   | [-]     | bool   | flag whether to enable geometric pull out               | true          |
+| divide_pull_out_path   | [-]     | bool   | flag whether to devide arc paths.　The path is assumed to be divided because the curvature is not continuous. But it requires a stop during the departure.                | false          |
```
Comment (satoshi-ota) [inline planning/behavior_path_planner/behavior_path_planner_pull_out-design.md:109]: typo

File: planning/behavior_path_planner/behavior_path_planner_pull_over-design.md [hunk 4]
```diff
@@ -0,0 +1,235 @@
+# PullOver Module
```
Comment (satoshi-ota) [inline planning/behavior_path_planner/behavior_path_planner_pull_over-design.md:1]:
```suggestion
# Pull Over design
```

File: planning/behavior_path_planner/behavior_path_planner_pull_over-design.md [hunk 5]
```diff
@@ -0,0 +1,235 @@
...
+| backward_parking_velocity                | [m/s] | double | velocity when backward parking                                            | -1.38         |
+| backward_parking_lane_departure_margin   | [m/s] | double | lane departure margin for front right corner of ego-vehicle when backward | 0.0           |
+
+### freespace parking
+
+If the vehicle gets stuck with `lane_parking`, run `freespace_parking`.
+To run this feature, you need to set `parking_lot` to the map, `activate_by_scenario` of [costmap_generator](../costmap_generator/README.md) to `false` and `enable_freespace_parking` to `true`
+
+![pull_over_freespace_parking_flowchart](./image/pull_over_freespace_parking_flowchart.drawio.svg)
+\*Series execution with `avoidance_moudle` in the flowchart is under development.
```
Comment (satoshi-ota) [inline planning/behavior_path_planner/behavior_path_planner_pull_over-design.md:217]:
```suggestion
\*Series execution with `avoidance_module` in the flowchart is under development.
```

File: planning/behavior_path_planner/behavior_path_planner_pull_out-design.md (hunk 3 above)
Comment (kosuke55) [reply planning/behavior_path_planner/behavior_path_planner_pull_out-design.md]: fix in https://github.com/autowarefoundation/autoware.universe/pull/2948/commits/339db78f30baa093e2f67a5ef52797cf39c1c2d8

File: planning/behavior_path_planner/behavior_path_planner_pull_out-design.md (hunk 2 above)
Comment (kosuke55) [reply planning/behavior_path_planner/behavior_path_planner_pull_out-design.md]: https://github.com/autowarefoundation/autoware.universe/pull/2948/commits/339db78f30baa093e2f67a5ef52797cf39c1c2d8

**Review by TakaHoribe — APPROVED** [review]

**Pull request merged**